returnn/tf/layers/rec.py

```diff
@@ -4873,6 +4873,9 @@ def update_var():
         s = tf.cond(tf.equal(step, 0), lambda: tf.zeros(initial_shape), lambda: var.value())
         s.set_shape(shape_invariant)
         return s
+    elif initial_state == "placeholder":
+      ph = tf_compat.v1.placeholder(tf.float32, shape=shape_invariant, name="placeholder_%s" % key_name)
```

Comment:
Better use `with rec_layer.var_creation_scope():` as above to make sure the scope is correct.